ItemRack/ItemRackQueue.lua

```diff
@@ -20,16 +20,16 @@ function ItemRack.ProcessAutoQueue(slot)
 	end
 
 	local start,duration,enable = GetInventoryItemCooldown("player",slot)
-	local timeLeft = GetTime()-start
+	local timeLeft = math.max(start + duration - GetTime(),0)
 	local baseID = ItemRack.GetIRString(GetInventoryItemLink("player",slot),true,true)
 	local icon = _G["ItemRackButton"..slot.."Queue"]
 
 	if not baseID then return end
 
 	local buff = GetItemSpell(baseID)
 	if buff then
-		if AuraUtil.FindAuraByName(buff,"player") or (start>0 and (duration-timeLeft)>30 and timeLeft<1) then
```

Comment:
Why don't we need "or (start>0 and (duration-timeLeft)>30 and timeLeft<1)" anymore?

Reply:
I was trying to work out what the intent of this line is before, but it made no sense to me.
start is >0 if the trinket has been used and cd is active, it basically has the time of trigger as the value
duration is the time of the cd in seconds, e.g. 2min cd trinket has this as a fixed value of 120, but this is only 120 if start>0
timeLeft (GetTime()-start) was calculated in a strange way that gave negative numbers when off CD, or it counted up from 0 when the item was triggered, until it's off cd.
so using that as an example, let's see what the values look like at 0 seconds after you use an item:
GetTime() = 1250
start = 1250
duration = 120
timeLeft = 0
start>0 and (duration-timeLeft)>30 and timeLeft<1
start>0 and (-120)>30 and 0<1
true and false and true

at 5 seconds after trigger:
GetTime() = 1255
start = 1250
duration = 120
timeLeft = 5
start>0 and (115)>30 and 5<1
true and true and false

item off cd:
GetTime() = 1370
start = 0
duration = 0
timeLeft = -1370
start>0 and (0 - -1370)>30 and -1370 < 1
false and true and false

As the 3 examples above show, the 3 conditions can never be all true at the same time, so it will never pass. This makes the check useless.

Reply:
makes sense, thanks for the detailed explanation.